src/test/java/org/jvnet/hudson/plugins/seleniumhtmlreport/SeleniumHtmlReportTest.java

```diff
@@ -0,0 +1,128 @@
+package org.jvnet.hudson.plugins.seleniumhtmlreport;
+
+import hudson.FilePath;
+import jenkins.tasks.SimpleBuildStep;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.steps.CoreStep;
+import org.jenkinsci.plugins.workflow.steps.StepConfigTester;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.*;
+
+public class SeleniumHtmlReportTest {
+
+    @Rule
+    public JenkinsRule story = new JenkinsRule();
+
+    @Test
+    public void configRoundTrip() throws Exception {
+        SeleniumHtmlReportPublisher reportPublisher = new SeleniumHtmlReportPublisher(".", false);
+        CoreStep step = new CoreStep(reportPublisher);
+        step = new StepConfigTester(story).configRoundTrip(step);
+        SimpleBuildStep delegate = step.delegate;
+
+        assertTrue(String.valueOf(delegate), delegate instanceof SeleniumHtmlReportPublisher);
```

Comment:
BTW you can get fancy with Hamcrest and use `assertThat(delegate, isA(SeleniumHtmlReportPublisher.class))` but I am not convinced the learning curve pays enough benefits.